src/Traversals/dfs.jl

```diff
@@ -2,52 +2,67 @@ import Base.showerror
 struct CycleError <: Exception end
 Base.showerror(io::IO, e::CycleError) = print(io, "Cycles are not allowed in this function.")
 
-struct DFS <: TraversalAlgorithm end
+struct DepthFirst <: TraversalAlgorithm
```

Comment:
```suggestion
struct DepthFirst{F <: Function} <: TraversalAlgorithm
```

with the field of type F below, allows for type stability

Reply:
we're not seeing any type instability right now. How does parameterizing a structure help with this?